pkg/controller/workload/workload_processor.go

```diff
@@ -609,6 +609,10 @@ func (p *Processor) storeServiceData(serviceName string, waypoint *workloadapi.G
 		newValue.ServicePort[i] = nets.ConvertPortToBigEndian(port.ServicePort)
 		if strings.Contains(serviceName, "waypoint") {
 			newValue.TargetPort[i] = nets.ConvertPortToBigEndian(KmeshWaypointPort)
+		} else if port.TargetPort == 0 {
+			// NOTE: Target port could be unset in servicen entry, in which case it should
+			// be consistent with the Service Port.
+			newValue.TargetPort[i] = nets.ConvertPortToBigEndian(port.ServicePort)
```

Comment:
I can see istio has set the traget port

(https://github.com/istio/istio/blob/081a061e195f6538df2b1dbb610217aa3fe596b6/pilot/pkg/serviceregistry/kube/controller/ambient/services.go#L145-L153)


Which istio version are you using

Reply:
Looks this is set since istio 1.23

Reply:
oh, I tested it with istio 1.22